demo/src/App.tsx

```diff
@@ -2,38 +2,67 @@ import React, { useState } from "react";
 import SvgDisplay from "./SvgDisplay";
 import MemoryModelsUserInput from "./MemoryModels";
 import { ErrorBoundary } from "react-error-boundary";
+import DownloadJSONButton from "./DownloadJSONButton";
+import DownloadSVGButton from "./DownloadSVGButton";
+import { Alert, Stack } from "@mui/material";
 
 export default function App() {
-    const [formData, setFormData] = useState("");
+    const [textData, setTextData] = useState("");
     const [jsonResult, setJsonResult] = useState(null);
+    const [svgResult, setSvgResult] = useState(null);
+    const [showFailureBanner, setShowFailureBanner] = useState(false);
 
-    const onSubmit = async (event) => {
+    const onSubmit = (event, data) => {
         event.preventDefault();
         try {
-            setJsonResult(JSON.parse(formData));
+            setJsonResult(JSON.parse(data));
+            setShowFailureBanner(false);
         } catch (error) {
             console.error(`Error parsing inputted JSON: ${error.message}`);
+            setShowFailureBanner(true);
             setJsonResult(null);
+            setSvgResult(null);
         }
     };
+    const onTextDataSubmit = (event) => {
+        onSubmit(event, textData);
+    };
 
     return (
         <>
+            {showFailureBanner && (
+                <Alert severity="error" data-testid="json-parse-alert">
+                    Failed to parse JSON. Please input valid JSON and re-draw
+                    diagram
+                </Alert>
+            )}
             <MemoryModelsUserInput
-                formData={formData}
-                setFormData={setFormData}
-                onSubmit={onSubmit}
+                textData={textData}
+                setTextData={setTextData}
+                onTextDataSubmit={onTextDataSubmit}
             />
             <section>
                 <h2>Output</h2>
                 <ErrorBoundary
                     fallback={
                         <div data-testid="svg-display-error-boundary">
-                            Something went wrong
+                            This is valid JSON but not valid Memory Models JSON.
+                            Please refer to the repo for more details.
                         </div>
                     }
+                    key={jsonResult}
                 >
-                    <SvgDisplay jsonResult={jsonResult} />
+                    <Stack
+                        direction={{ xs: "column", sm: "row" }}
+                        spacing={{ xs: 1, sm: 2, md: 4 }}
+                    >
+                        <DownloadJSONButton jsonResult={jsonResult} />
```

Comment:
In terms of UI, what I'd like to see is that above the JSON text input:

- A file input on the top-left
- The "(Re)Apply..." button immediately to the right of it (not below it)
- The "Download This JSON" button on the top-right

Reply:
![image](https://github.com/david-yz-liu/memory-models-rough/assets/67441706/ebb3acc9-c95b-4602-8426-0e19c84ff576)